common/src/logger.rs

```diff
@@ -1,71 +1,45 @@
 
 /// The global logger instance used for the `log` crate.
 pub static LOGGER: OnceCell<LockedLogger> = OnceCell::uninit();
 
-/// A [`Logger`] instance protected by a spinlock.
-pub struct LockedLogger(Spinlock<Logger>);
-
-/// Additional vertical space between lines
-const LINE_SPACING: usize = 2;
-/// Additional horizontal space between characters.
-const LETTER_SPACING: usize = 0;
-
-/// Padding from the border. Prevent that font is too close to border.
-const BORDER_PADDING: usize = 1;
-
-/// Constants for the usage of the [`noto_sans_mono_bitmap`] crate.
-mod font_constants {
-    use super::*;
-
-    /// Height of each char raster. The font size is ~0.84% of this. Thus, this is the line height that
-    /// enables multiple characters to be side-by-side and appear optically in one line in a natural way.
-    pub const CHAR_RASTER_HEIGHT: RasterHeight = RasterHeight::Size16;
-
-    /// The width of each single symbol of the mono space font.
-    pub const CHAR_RASTER_WIDTH: usize = get_raster_width(FontWeight::Regular, CHAR_RASTER_HEIGHT);
-
-    /// Backup character if a desired symbol is not available by the font.
-    /// The '�' character requires the feature "unicode-specials".
-    pub const BACKUP_CHAR: char = '�';
-
-    pub const FONT_WEIGHT: FontWeight = FontWeight::Regular;
-}
-
-/// Returns the raster of the given char or the raster of [`font_constants::BACKUP_CHAR`].
-fn get_char_raster(c: char) -> RasterizedChar {
-    fn get(c: char) -> Option<RasterizedChar> {
-        get_raster(
-            c,
-            font_constants::FONT_WEIGHT,
-            font_constants::CHAR_RASTER_HEIGHT,
-        )
-    }
-    get(c).unwrap_or_else(|| get(BACKUP_CHAR).expect("Should get raster of backup char."))
-}
+/// A [`FrameBufferWriter`] instance protected by a spinlock.
+pub struct LockedLogger(
+    Spinlock<FrameBufferWriter>,
+    Spinlock<SerialPort>,
+    LoggerStatus,
+    LoggerStatus,
+);
 
 impl LockedLogger {
     /// Create a new instance that logs to the given framebuffer.
-    pub fn new(framebuffer: &'static mut [u8], info: FrameBufferInfo) -> Self {
-        LockedLogger(Spinlock::new(Logger::new(framebuffer, info)))
+    pub fn new(
+        framebuffer: &'static mut [u8],
+        info: FrameBufferInfo,
+        frame_buffer_logger_status: LoggerStatus,
+        serial_logger_status: LoggerStatus,
+    ) -> Self {
+        LockedLogger(
+            Spinlock::new(FrameBufferWriter::new(framebuffer, info)),
+            Spinlock::new(SerialPort::new()),
+            frame_buffer_logger_status,
+            serial_logger_status,
+        )
     }
 
     /// Force-unlocks the logger to prevent a deadlock.
     ///
     /// ## Safety
     /// This method is not memory safe and should be only used when absolutely necessary.
     pub unsafe fn force_unlock(&self) {
-        unsafe { self.0.force_unlock() };
+        unsafe {
+            self.0.force_unlock();
+            self.1.force_unlock();
+        };
```

Comment:
With the above changes, we could use `if let` here:

```suggestion
        if let Some(framebuffer) = &mut self.framebuffer {
            unsafe { framebuffer.force_unlock() };
        }
        if let Some(serial) = &mut self.serial {
            unsafe { serial.force_unlock() };
        }
```